azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/action/CaptionsAction.kt

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.redux.action
+
+import com.azure.android.communication.ui.calling.models.CallCompositeCaptionsType
+import com.azure.android.communication.ui.calling.redux.state.CaptionsStatus
+
+internal sealed class CaptionsAction : Action {
+    class StartRequested(val language: String?) : CaptionsAction()
+    class StopRequested : CaptionsAction()
+    class UpdateStatus(val status: CaptionsStatus) : CaptionsAction()
+    class SetSpokenLanguageRequested(val language: String) : CaptionsAction()
+    class SpokenLanguageChanged(val language: String) : CaptionsAction()
+    class SetCaptionLanguageRequested(val language: String) : CaptionsAction()
+    class CaptionLanguageChanged(val language: String) : CaptionsAction()
+    class TypeChanged(val type: CallCompositeCaptionsType) : CaptionsAction()
+    class IsTranslationSupportedChanged(val isSupported: Boolean) : CaptionsAction()
+    class SupportedSpokenLanguagesChanged(val languages: List<String>) : CaptionsAction()
+    class SupportedCaptionLanguagesChanged(val languages: List<String>) : CaptionsAction()
+    class ShowCaptionsOptions : CaptionsAction()
```

Comment:
Show/Hide should probably be in navigation actions if involves showing/hiding a drawer.

Reply:
Hi Adam, NavigationActions vs state is managed by NavigationManager to exit composite or moving to specific screen. 
I think as captions have state and actions for self, I think we can keep in captions state and actions.